reapable/reapables.go

```diff
@@ -72,42 +82,30 @@ func (rs *Reapables) Get(region Region, id ID) (Reapable, error) {
 	return r, NotFoundError{fmt.Sprintf("Could not find resource %s in %s", id.String(), region.String())}
 }
 
+// Delete removes a Reapable from Reapables
 func (rs *Reapables) Delete(region Region, id ID) {
-	rs.RLock()
-	defer rs.RUnlock()
+	rs.Lock()
+	defer rs.Unlock()
 	delete(rs.storage[region], id)
 }
 
-type ReapableContainer struct {
-	Reapable
-	region Region
-	id     ID
-}
-
-func (r *ReapableContainer) Region() Region {
-	return r.region
-}
-
-func (r *ReapableContainer) ID() ID {
-	return r.id
-}
-
-func (rs *Reapables) Iter() <-chan ReapableContainer {
-	ch := make(chan ReapableContainer)
-	go func(c chan ReapableContainer) {
-		rs.Lock()
-		defer rs.Unlock()
-		for region, regionMap := range rs.storage {
-			for id, r := range regionMap {
-				c <- ReapableContainer{r, region, id}
+// Iter returns an iterable channel of Reapables
+func (rs *Reapables) Iter() <-chan Reapable {
+	ch := make(chan Reapable)
+	go func(c chan Reapable) {
+		rs.RLock()
+		defer rs.RUnlock()
```

Comment:
A quick way to fix this would be to make the channel queue large enough to cover the map.
```
items := 0
for _, regionMap := range rs.storage {
 items += len(regionMap)
}
make(chan Reapable, items)
```